### ISSUE TITLE
The IO test files are too large

### ISSUE THREAD

**Issue by francisco-dlp** [hyperspy/hyperspy#2048]
The file size of the ``IO`` test folder is 88.2 MB. As a consequence, the file size of HyperSpy v1.4 source distribution is 67.7 MB and hits the PyPI's file size limit, see https://github.com/pypa/warehouse/issues/4659. This is currently holding the v1.4 release.

The bigger offenders are:

* ``edax_files.zip`` 36.5 MB
* ``bruker_data/`` 12.8 MB
* ``FEI_new/`` 10.3 MB
* ``emd_files/`` 9.3 MB
* ``dm4_1D_data/`` 4.5 MB
* ``tiff_files/`` 3.5 MB

Hopefully the PyPI maintainers will increase the file size limit of our project so that we can go on with the release. However, regardless, we should reduce the file size of the test files when possible. Otherwise this is going to remain an issue for the planned separate IO library #1978.

**Comment by magnunor** [comment]
From a longer term perspective, one solution could be to compress all the files, and use the "decompress" trick in the test themselves. Simply compressing the IO-test folder reduces it from 88 MB to 57 MB.

We could also try using files which only have zero values, which would make the compression much more efficient. For example, in `fpd` 33 MB of Merlin binary files (mib), is reduced to 80 kB, due to the files being only zeros.

**Comment by ericpre** [comment]
A short term workaround is to remove some of the large files and skip the
corresponding tests just for the release... Not great but this would do for
this release.

**Comment by ericpre** [comment]
I meant remove from packaging only, not from the source code.

>

**Comment by francisco-dlp** [comment]
@magnunor, thanks for the suggestion. Unfortunately compressing the files doesn't help with this because the source distribution is itself compressed. I've tried compressing all the files with zip and creating a new sdist and I get exactly the same file size. Using files that only have zero values should help, but we won't be able to cover all our test cases with zero-filled files.

@ericpre, we could remove files and skip tests, but as pypi is the primary archiving repository of Python packages, I would consider that an action of last resort (i.e. something we could reluctantly do if they don't increase the file size limit of our project as long as nobody comes with a better alternative). So far we haven't heard from PyPA.

**Comment by sem-geologist** [comment]
@francisco-dlp , I think I had already discussed previously against double and triple compressions as that actually 1. increase the size and 2. puts overhead. And I am looking to you EDAX_test files.

As I see my Bruker test files is biggest offender in above list, but You probably had forgot EDAX files, which come without folder but with compressed to 34.8MB zip file (as it is allready compressed no more compression from it in final pypy; contrary to that, BCF files are compressed down to 7MB).
That does not mean that I should get lazy and not look over and review and replacing some of bruker files with much smaller files (e.g. with smaller pixel*pixel mappings). But I think EDAX_files.zip is the real offender here now.

**Comment by sem-geologist** [comment]
@jat255 edax_files.zip needs to be so huge?

**Comment by sem-geologist** [comment]
@magnunor , I placed cons against onion-zlibing here #1838.

**Comment by francisco-dlp** [comment]
@sem-geologist, good and useful catch! Somehow I missed the elephant in the room!

**Comment by francisco-dlp** [comment]
If #2049 sounds like a good solution to the issue, I'll merge it and close pypa/warehouse#4659, but not this issue as #2049 is only a temporary solution.

**Comment by francisco-dlp** [comment]
What do you think?

**Comment by jat255** [comment]
As we mentioned, #2049 is only a temporary solution. For the Edax case, I am not aware of a method to get a significantly smaller filesize for a map.

I see a couple of options:
1) Host the test files somewhere else and download as needed within the test scripts (not ideal)
1) (Sort of a follow-on of number 1) Set up another repository solely as a large file-storage for HyperSpy, so it will always be accessible (as long as Github is), and won't be pulled into the bundle. This would allow us to version the files and make sure the latest versions are always available. There's also [Git LFS](https://git-lfs.github.com/), although I'm not sure if this is a correct use case for it or not, since I've never used it before.
1) I guess a third option could be to move to tests to another repository entirely, but that's more extreme and I don't think necessary.

My vote would be for 2. Thoughts?

**Comment by sem-geologist** [comment]
@jat255, I think there should be much more easier and simpler solution. Why EDAX files need to be so huge, why not redo them for test to as minimal as possible? I don't think tests benefit from UHD, 4k (I am a bit exaggerating) like dimensions, while 50x50 pixels and even smaller dimensions would do the same job for testing. As an example, for bruker testing I have made 16x16 hypermaps. Which takes very little space.

**Comment by jat255** [comment]
At the time of writing the edax reader, I did not know how (or if) it was possible to collect smaller files via the EDAX software (and I no longer have access to an EDAX detector setup on which to test). 

One approach that I've used in some other work (and I think @magnunor mentioned this too) is to zero-out the bytes in the parts of the files that are used for raw data storage. This allows the compressed files to be very small, as only the metadata is actually stored. There is an impact on file storage while the tests are running, but I don't think this is an issue on most development machines.

This would be an effective, and not-too-difficult-to-implement solution. The only downside is that the files are not "raw" as collected by the software, but I think with careful implementation, this is okay.

---

The EDAX files were already removed from the distribution, in lieu of a web-download solution [here](https://github.com/hyperspy/hyperspy/blob/81c226ac4fc6a32e09022ce0ce2b29bb90308269/hyperspy/tests/io/test_edax.py#L28-L42). We will need someone with an EDAX system that can test out creating smaller files if we want to "properly" fix this
